### PR TITLE
feat: indent code dynamically

### LINKED ISSUE #46: feat: editor tabSize should depend on the language used
When using a code editor, the `tabSize` parameter should take into account the language used.

A helper function of a mapping from languages to tabSize values could be helpful.

### PR DESCRIPTION
**Pull request by juancarlosfarah**
Indent code according to each programming language's best practices.

closes #46

### CODE REVIEW

**Comment by juancarlosfarah** [comment]
@spaenleh, no rush to review this as I have already deployed it. Just copying you for your awareness.

File: src/components/common/CodeLine.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import AddIcon from '@material-ui/icons/Add';
 import PropTypes from 'prop-types';
 import 'prismjs/themes/prism.css';
 import { MessageOutlined } from '@material-ui/icons';
+import { programmingLanguageSettings } from '../../constants/programmingLanguages';
```
Comment (juancarlosfarah) [inline src/components/common/CodeLine.js:7]: Just testing a comment here.

Comment (juancarlosfarah) [reply src/components/common/CodeLine.js:7]: And here is a reply.

**Comment by juancarlosfarah** [comment]
@spaenleh, I'm going to merge this unless you have any questions / comments / suggestions.

**Review by spaenleh — APPROVED** [review]
Looks good to me ! 🚀

**Pull request merged**